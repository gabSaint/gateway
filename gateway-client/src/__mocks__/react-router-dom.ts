import React, { FunctionComponent } from "react";

const useParams = jest.fn();
const useHistory = jest.fn();

const Link: FunctionComponent = ({ children }) =>
  React.createElement("div", null, children);

export { useParams, useHistory, Link };
