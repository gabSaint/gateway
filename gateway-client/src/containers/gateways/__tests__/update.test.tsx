import React from "react";
import { ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import renderer from "react-test-renderer";
import { useParams } from "react-router-dom";

import UpdateGateways from "../update";

describe("UpdateGateways", () => {
  let wrapperUpdateGateways: ShallowWrapper | ReactWrapper;

  beforeEach(() => {
    // @ts-ignore
    useParams.mockImplementationOnce(() => ({ id: "1" }));

    wrapperUpdateGateways = shallow(<UpdateGateways />);
  });

  it("renders correctly", () => {
    // @ts-ignore
    useParams.mockImplementationOnce(() => ({ id: "1" }));

    const tree = renderer.create(<UpdateGateways />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders delete button", () => {
    const deleteButtons = wrapperUpdateGateways.find(
      "[data-test='form-button-delete']"
    );
    expect(deleteButtons).toHaveLength(1);
  });

  it("renders gateway form", () => {
    const form = wrapperUpdateGateways.find("GatewayForm");
    expect(form).toHaveLength(1);
  });
});
