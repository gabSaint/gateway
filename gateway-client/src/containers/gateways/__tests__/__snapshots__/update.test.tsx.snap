// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateGateways renders correctly 1`] = `
<div
  className="contain-row"
>
  <h2>
    Edit Gateway
  </h2>
  <button
    className="negative"
    data-test="form-button-delete"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
